src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ProblemMessageControllerTest.java

```diff
@@ -0,0 +1,205 @@
+import uk.gov.hmcts.reform.wacaseeventhandler.entity.MessageState;
+import uk.gov.hmcts.reform.wacaseeventhandler.services.DeadLetterQueuePeekService;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.Collectors;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc(addFilters = false)
+@ActiveProfiles(profiles = {"db", "integration"})
+@TestPropertySource(properties = {"azure.servicebus.enableASB=true",
+    "azure.servicebus.connection-string="
+    + "Endpoint=sb://REPLACE_ME/;SharedAccessKeyName=REPLACE_ME;SharedAccessKey=REPLACE_ME",
+    "azure.servicebus.topic-name=test",
+    "azure.servicebus.subscription-name=test",
+    "azure.servicebus.ccd-case-events-subscription-name=test",
+    "azure.servicebus.retry-duration=2"})
+class ProblemMessageControllerTest {
+    protected static final ObjectMapper OBJECT_MAPPER = new ObjectMapper()
+        .setPropertyNamingStrategy(PropertyNamingStrategies.UPPER_CAMEL_CASE)
+        .registerModule(new JavaTimeModule())
+        .registerModule(new Jdk8Module());
+
+    public static final String S2S_TOKEN = "Bearer s2s token";
+    public static final String CASE_REFERENCE = "some case reference";
+    public static final LocalDateTime EVENT_TIME_STAMP = LocalDateTime.now();
+    public static final LocalDateTime OLD_EVENT_TIME_STAMP = LocalDateTime.now().minusHours(2);
+
+    @MockBean
+    private AuthTokenGenerator authTokenGenerator;
+
+    @MockBean
+    private DeadLetterQueuePeekService deadLetterQueuePeekService;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private LaunchDarklyFeatureFlagProvider launchDarklyFeatureFlagProvider;
+
+    @BeforeEach
+    void setUp() {
+        when(launchDarklyFeatureFlagProvider.getBooleanValue(any(), any())).thenReturn(true);
+        when(authTokenGenerator.generate()).thenReturn(S2S_TOKEN);
+    }
+
+    @NotNull
+    private String randomMessageId() {
+        return "messageId_" + ThreadLocalRandom.current().nextLong(1000000);
+    }
+
+    @Test
+    void should_return_a_ready_message_when_ready_message_timestamp_is_old_than_one_hour() throws Exception {
```

Comment:
small adjustment to name. older can replace old to emphasise test meaning

Reply:
done